### ISSUE TITLE
`phpcbf` FAILED TO FIX

### ISSUE THREAD

**Issue by nguyenk** [squizlabs/PHP_CodeSniffer#3023]
Hello !

I'm running into a problem  : 
![image](https://user-images.githubusercontent.com/2227554/87863590-bf302880-c95c-11ea-83e7-7bb9eea621f4.png)

Please find attached the file that messes up.
[MappingUtils.zip](https://github.com/squizlabs/PHP_CodeSniffer/files/4942853/MappingUtils.zip)

I you need any more info, please ask !

**Comment by nguyenk** [comment]
Seems I'm running into an inifite loop - running with `-vv` says :
`	*** Reached maximum number of loops with 73 violations left unfixed ***`

**Comment by jrfnl** [comment]
Hi @nguyenk It would be helpfu if you could provide the following additional information:
* Which PHPCS version are you on ?
* Running on which PHP version ?
* Using which standard ? Or if using a custom ruleset - the ruleset ?

Also, please try and identify the smallest code snippet with which to reproduce the fixer conflict.

**Comment by nguyenk** [comment]
Hi !

 * PHPCS version is 3.5.4
 * PHP version 7.4.7
 * I dont know about the standards I'm using, how can I check ?

I tried removing code blocks, but this is very weird, removing some code blocks make the fix pass, but removing the exact opisit too. So I'm not able to identify which code portions are causing the problem.

**Comment by jrfnl** [comment]
> I dont know about the standards I'm using, how can I check ?

How are you running PHPCS ? What is the command you are using with all the parameters ?

If the command contains `--standard=....`, what's listed there is the standard you are using.

If the command doesn't have the `--standard=...` parameter, look for a `[.]phpcs.xml[.dist]` file in the directory from which you are running PHPCS . If you can't find such a file, keep looking in the directories above the current directory to see if any of them contain it.
If you find the file, open it up and see what it says inside, there should be at least one `<rule ref="...."/>` directive.

If neither of the above yields any results, run `phpcs --config-show` and check if a `default_standard` is listed. If it is, that's what you are using.

If even that doesn't give you a standard, you will be using the `PEAR` standard which is the system default.

**Comment by gsherwood** [comment]
I haven't been able to replicate this problem with any of the included standards, so I will really need to know that standard to help.

Can you please run PHPCS over this code with the `--report=diff -vvv` command line arguments.
For example: `phpcs path/to/MappingUtils.php --report=diff -vvv`

This will output a lot of information, but it will tell me exactly what sniffs are causing the problem.

**Comment by nguyenk** [comment]
Hello, 

First of all, thank you for trying to help, and sorry for the delay, I'm in a big rush right now :sweat_smile: 

Please fin attached the `phpcs.xml.dist` file, it seems I'm running the following standards : 
 * doctrine
 * slevomat

I also atatched the `MappingUtils.php` file again and the outup of `phpcs path/to/MappingUtils.php --report=diff -vvv` :
[phpcs.zip](https://github.com/squizlabs/PHP_CodeSniffer/files/4957871/phpcs.zip)

Finally, I think I need to let you know I have been able to fix the issue by commenting half of he file, running `phpcbf` once, then uncommenting that previously commented part, and running `phpcbf` again.

Maybe this was just a 'max Depth' problem (there was 62 errors, I'm not so proud :laughing: ) ?

**Comment by gsherwood** [comment]
Thanks a lot for providing that log.

It looks like all the changes are being made by sniffs from the Slevomat coding standard, so I don't think this is something I can resolve in this project.

As best as I can tell, the problem is this error:
```
E: [Line 64] Function \in_array() should not be referenced via a fully qualified name, but via a use statement. 
(SlevomatCodingStandard.Namespaces.ReferenceUsedNamesOnly.ReferenceViaFullyQualifiedName)
```
adds a use statement to the bottom of the user list of the file: `use function in_array;`

That then generates:
```
E: [Line 16] Use statements should be sorted alphabetically. The first wrong one is in_array. 
(SlevomatCodingStandard.Namespaces.AlphabeticallySortedUses.IncorrectlyOrderedUses)
```

But the fix tries to touch the same tokens again and it starts to conflict.

All this needs to be resolved inside that project, so I'm going to close this issue now that the problem is clear. I recommend opening an issue at that project and see if they can resolve it: https://github.com/slevomat/coding-standard

**Comment by nguyenk** [comment]
Hi @gsherwood thanks a lot !

I'll open the issue right away.

Have a very good evening :surfing_man: !

**Comment by realjjaveweb** [comment]
Maybe this will help someone: if you're also getting this unfixable thing on otherwise php-valid file and you have debugged up to the `Reached maximum number of loops with...` point and you don't see any visible issues - try to think what could cause a conflict in formatting - _something where fixer could not decide how to deal with it_ - in my case I had one file that had following syntax:

```
<div class="wrapper" <?php if (!empty($wrapperConfig['add_id'])) {
    echo ' id="' . \htmlspecialchars($wrapperConfig['add_id']) . '"';
} ?>>
    <?php if (!finished()) { ?>
        ...
    <?php } else { ?>
        ...
    <?php } ?>
</div>
<?php
```

The fixer couldn't understand the nested partial wrapping of the first if, so all I did was change the 1st 3 lines to:
 ```
<div class="wrapper" <?php
if (!empty($wrapperConfig['add_id'])) {
    echo ' id="' . \htmlspecialchars($wrapperConfig['add_id']) . '"';
}
?>>
```
I also tried to put it in just 1 line and that also worked but ended up with a weird shifting formattnig on 2nd line.
```
<div class="wrapper" <?php if (!empty($wrapperConfig['add_id'])) { echo ' id="' . \htmlspecialchars($wrapperConfig['add_id']) . '"'; } ?>>
```

**Comment by lauripiisang** [comment]
Had the same issue in a project using quite old versions, I had a closure with odd phpdoc, like
```php
$collection->first(/**
  * @param Type|Type2 $elem
 */ function($elem) use($otherVar) {
```

and fix was to separate tokens to a more reasonable structure, like
```php
$collection->first(
  function($elem) use($otherVar) {
    /** @var Type|Type2 $elem */
```
Investigated with the command pointed out above (`vendor/bin/phpcbf <myfile> --report=diff -vv`)
